### ISSUE TITLE
QA Notice when building for gentoo

### ISSUE THREAD

**Issue by ghost** [brainpower/bp-nfoview#3]
Hello.
I got this notice when building on gentoo linux.

 * QA Notice: This package installs one or more .desktop files that do not
 * pass validation.
 * 
 * 	/usr/share/applications/bp-nfoview.desktop: warning: value "Application;" for key "Categories" in group "Desktop Entry" contains a deprecated value "Application"
 *

When i build and install bp-nfoview. I got a new dir in my applicationsmeny in my plasma 5 desktop, called lost & found. 
I think it would be better to put it in utilities instead.

You find an gentoo ebuild file there https://github.com/fhede/Gentoo-misc/tree/master/ebuilds

Thanks for a nice tool.

**Comment by brainpower** [comment]
Thanks for telling, I probably wouldn't had touched the desktop file for another 5 years or so. ^^

I double checked with https://standards.freedesktop.org/menu-spec/latest/apa.html and agree with you: Let's put it into Utility.

**Comment by brainpower** [comment]
@fhede 
Could you tell me the the commands I could add to the Install section of the README on how gentoo users could install this using the ebuild?
It'd be nice to have a similar section to the Arch Linux one. ;)

**Comment by brainpower** [comment]
oh, and do those cmake-utils set an appropriate CMAKE_BUILD_TYPE?
If not, you should probably set it to Release or RelWithDebInfo.
USE_QT5 defaults to ON and I don't plan on changing that. (At least not until Qt6)

**Comment by ghost** [comment]
Hello brainpower.

I  probably give you to much info here, but that 's just to show you the 
similarities to arch linux.

In gentoo they use a similar way like AUR for arch linux.

And that is called  a "layman overlay" in there people put ebuilds that 
are not in offical repos.

https://wiki.gentoo.org/wiki/Layman

https://wiki.gentoo.org/wiki/Overlay


And then gentoo have something they call  "local overlay" that is what i 
use for your tool,and that is like create an pkgbuild

and make an makepkg -si in arch linux.

https://wiki.gentoo.org/wiki/Custom_repository


--install guide -----------------------------------

How to install bp-nfoview in gentoo linux.

Create an local overlay and download the ebuild.

mkdir -p /usr/local/portage/app-text/bp-nfoview

cp my /home/user/bp-nfoview-2.0.ebuild  
/usr/local/portage/app-text/bp-nfoview/bp-nfoview-2.0.ebuild

chown -R portage:portage /usr/local/portage

pushd /usr/local/portage/app-text/bp-nfoview

repoman manifest

popd


install commands.

emerge --sync

emerge --ask --oneshot app-text/bp-nfoview

install guide finish--------------------------------------------------------


If you wonder why i called it app-text/bp-nfoview.

That's because i follow the same dirs like this similar tool in official 
repos.

https://gpo.zugaina.org/app-text/nfoview


The ebuild i created i see that i put version 1.0 but should be 2.0

in the top of the ebuild you can change to 2.0 there and call the ebuild 
2.0 instead.

[[ ${PV}=*1.0*]] &&inherit git-r3

I updated bp-nfoview with the new utilities dir and that looks much better.

thanks

fhede



Den 2017-02-27 kl. 18:16, skrev brainpower:

**Comment by ghost** [comment]
I followed this gentoo manual 
https://devmanual.gentoo.org/eclass-reference/cmake-utils.eclass/

*CMAKE_BUILD_TYPE*?=/Gentoo/
Set to override default CMAKE_BUILD_TYPE. Only useful for packages known 
to make use of "if (CMAKE_BUILD_TYPE MATCHES xxx)". If about to   be set 
- needs to be set before invoking cmake-utils_src_configure. You usualy 
do *NOT* want nor need to set it as it pulls CMake default build-type 
specific compiler flags overriding make.conf

And what i understand of it, is that everything should be set with 
cmake-utils.

fhede

Den 2017-02-27 kl. 18:59, skrev brainpower:

**Comment by brainpower** [comment]
Well, if setting it explicitly is not needed, then there's no need to do that. Just wanted to make sure it's not set to Debug. ;)

I've wanted to, but did not yet try gentoo yet and I only heard of a few things:
There's a tool named emerge, there are overlays and use flags. And that's it.
So thanks for explaining.
